Sync Gateway CBG-365 - Update logging config for SG 2.1.0+ (#126)
couchbase/docker: community/sync-gateway/2.1.0/README.md

This README will guide you through running Couchbase Sync Gateway with Docker Containers.

[Sync Gateway](https://www.couchbase.com/products/sync-gateway) is a horizontally scalable web server that securely manages the access control and synchronization of data between [Couchbase Lite](https://www.couchbase.com/products/lite) and [Couchbase Server](https://www.couchbase.com/products/server).

> For configuration references, or additional information about anything described below, visit the [Sync Gateway documentation site](https://docs.couchbase.com/sync-gateway/2.5/index.html).

Reviewer: ```suggestion
For configuration references, or additional information about anything described below, visit the [Sync Gateway documentation site](https://docs.couchbase.com/sync-gateway/current/index.html).
```

For additional questions and feedback, please visit the [Couchbase Forums](https://forums.couchbase.com/c/mobile/sync-gateway) or [Stack Overflow](https://stackoverflow.com/questions/tagged/couchbase+couchbase-sync-gateway).

# QuickStart with Sync Gateway and Docker

## Running Sync Gateway with Docker

```
$ docker run -d --name sgw -p 4984:4984 couchbase/sync-gateway
```

> At this point you should be able to send a HTTP request against Sync Gateway public port `4984` using curl:

Reviewer 2: ```suggestion
At this point you should be able to send a HTTP request to the Sync Gateway public port `4984` using curl:
```

2019-05-14T12:59:22.418Z [INF] Logging: Files to /var/log/sync_gateway
2019-05-14T12:59:22.418Z [INF] Logging: Console level: info
2019-05-14T12:59:22.418Z [INF] Logging: Console keys: [HTTP]
etc ...
```

# Admin Port

By default, port `4985`, which is the Sync Gateway Admin port, is only accessible via localhost for security purposes. This means that it's only accessible *from within the container*.

To access it from within the container, you can get a bash shell on the running container and then use curl to connect to the admin port as follows:

**Step - 1 :** Get access to a shell inside the container running Sync Gateway:

`$ docker exec -ti sgw bash`

**Step - 2 :** Run curl from container shell (indicated by `#` prompt):

```
# curl http://localhost:4985
{"ADMIN":true,"couchdb":"Welcome","vendor":{"name":"Couchbase Sync Gateway","version":"2.5"},"version":"Couchbase Sync Gateway/2.5.0(271;bf3ddf6) EE"}
```

## Exposing admin port to host

Although not recommended for security reasons, if you need to expose the admin port to the host machine, you can do so with the following steps.

**Step - 1 :** Stop any currently running Sync Gateway containers:

> `docker stop sgw`.

Reviewer: ```suggestion
`docker stop sgw`
```

# Running with a Couchbase Server container

**Step - 1 :** Create a docker network called `couchbase`:

`$ docker network create --driver bridge couchbase`

**Step - 2 :** Run Couchbase Server in a docker container, and put it in the `couchbase` network.

`$ docker run --net=couchbase -d --name couchbase-server -p 8091-8094:8091-8094 -p 11210:11210 couchbase`

Now go to the Couchbase Server Admin UI on [http://localhost:8091](http://localhost:8091) and go through the Setup Wizard.

See [Couchbase Server on Dockerhub](https://hub.docker.com/r/couchbase/server/) for more info on this process.

**Step - 3 :** Create a configuration file as described in the above config section, and customise the server property:

```
{
  "logging": {
    "console": {
        "enabled": true,
        "log_level": "info",
        "log_keys": ["HTTP"]
    }
  },
  "databases": {
    "db": {
      "server": "http://couchbase-server:8091",
      "bucket": "default",
      "username": "Administrator",
      "password": "password",
      "users": { "GUEST": { "disabled": false, "admin_channels": ["*"] } }
    }
  }
}
```

> **Step - 3 :** Start a Sync Gateway container in the `couchbase` network and use the configuration file you just wrote:

Reviewer: ```suggestion
**Step - 4 :** Start a Sync Gateway container in the `couchbase` network and use the configuration file you just wrote:
```

`$ docker run --net=couchbase -p 4984:4984 -v /tmp:/tmp/config -d couchbase/sync-gateway /tmp/config/my-sg-config.json`

# Collecting logs via sgcollect_info

This section only applies if you need to run the `sgcollect_info` tool to collect Sync Gateway diagnostics for support.

**Step - 1 :** Run the following curl command against the admin port of Sync Gateway to run sgcollect_info and put the zip in your log file path.

`# curl -X POST http://localhost:4985/_sgcollect_info -H 'Content-Type: application/json' -d '{}'`

You can find more information about the parameters used in this request in the [sgcollect_info documentation](https://docs.couchbase.com/sync-gateway/current/admin-rest-api.html#/server/post__sgcollect_info).

# Licensing

Sync Gateway comes in 2 Editions: Enterprise Edition and Community Edition. You can find details on the differences between the 2 and licensing details on the [Sync Gateway Editions](http://developer.couchbase.com/documentation/server/4.5/introduction/editions.html) page.

> - **Enterprise Edition** -- free for development, testing and POCs. Requires a paid subscription for production deployment. Please refer to the [subscribe](https://www.couchbase.com/subscriptions-and-support) page for details on enterprise edition agreements.

Reviewer: ```suggestion
- **Enterprise Edition** -- free for development, testing and POCs. Requires a paid subscription for production deployment. Please refer to the [subscriptions](https://www.couchbase.com/subscriptions-and-support) page for details on enterprise edition agreements.
```

$ vi my-sg-config.json  # make edits
```

**Step - 2 :** Run Sync Gateway and use that configuration file:

`$ docker run -p 4984:4984 -d -v /tmp:/tmp/config couchbase/sync-gateway /tmp/config/my-sg-config.json`

## Using a URL

> Sync Gateway can also load it's configuration directly from a public URL.

Reviewer 2: ```suggestion
Sync Gateway can also load its configuration directly from a public URL.
```

**Step - 1 :** Upload a configuration file to a publicly available hosting site of your choice (Amazon S3, Github, etc)

**Step - 2 :** Then start Sync Gateway and give it the URL to the raw JSON data:

`$ docker run -p 4984:4984 -d couchbase/sync-gateway https://raw.githubusercontent.com/couchbase/sync_gateway/master/examples/basic-walrus-bucket.json`

# Running with a Couchbase Server container

**Step - 1 :** Create a docker network called `couchbase`:

`$ docker network create --driver bridge couchbase`

**Step - 2 :** Run Couchbase Server in a docker container, and put it in the `couchbase` network.

`$ docker run --net=couchbase -d --name couchbase-server -p 8091-8094:8091-8094 -p 11210:11210 couchbase`

Now go to the Couchbase Server Admin UI on [http://localhost:8091](http://localhost:8091) and go through the Setup Wizard.

See [Couchbase Server on Dockerhub](https://hub.docker.com/r/couchbase/server/) for more info on this process.

> **Step - 3 :** Create a configuration file as described in the above config section, and customise the server property::

Reviewer 2: ```suggestion
**Step - 3 :** Create a configuration file as described in the above config section, and customise the server property:
```

```
{
  "logging": {
    "console": {
        "enabled": true,
        "log_level": "info",
        "log_keys": ["HTTP"]
    }
  },
  "databases": {
    "db": {
      "server": "http://couchbase-server:8091",
      "bucket": "default",
      "username": "Administrator",
      "password": "password",
      "users": { "GUEST": { "disabled": false, "admin_channels": ["*"] } }
    }
  }
}
```

**Step - 3 :** Start a Sync Gateway container in the `couchbase` network and use the configuration file you just wrote:

`$ docker run --net=couchbase -p 4984:4984 -v /tmp:/tmp/config -d couchbase/sync-gateway /tmp/config/my-sg-config.json`

# Collecting logs via sgcollect_info

This section only applies if you need to run the `sgcollect_info` tool to collect Sync Gateway diagnostics for support.

**Step - 1 :** Run the following curl command against the admin port of Sync Gateway to run sgcollect_info and put the zip in your log file path.

`# curl -X POST http://localhost:4985/_sgcollect_info -H 'Content-Type: application/json' -d '{"upload": false, "customer": "your-company-name"}'`

> Set upload to `true` along with a customer name if you wish to automatically upload the collection to Couchbase Support.

Reviewer 2: ```suggestion

```

Tempted to cut this down a bit, and leave it to point to the main docs. Lowers the risk of random uploads/doesn't set the expectation that Support will engage any user.
